SQL/Archive/17.0/2016-08-19-IssueTracker.sql

```diff
@@ -0,0 +1,111 @@
+-- Add issues tab to the Loris Menu
+INSERT INTO `LorisMenu` (`Parent`, `Label`, `Link`, `Visible`, `OrderNumber`)
+VALUES (5, 'Issue Tracker', 'issue_tracker/', true, 8);
+
+
+-- Add user permissions
+-- Reporter
+INSERT INTO `permissions` (`code`, `description`, `categoryID`)
+VALUES ('issue_tracker_reporter', 'Can add a new issue, edit own issue, comment on all', 2);
+INSERT IGNORE INTO user_perm_rel (`userID`, `permID`) VALUES (
+  (SELECT ID FROM users WHERE UserID = 'admin'), (SELECT permID FROM permissions WHERE code = 'issue_tracker_reporter')
+);
+
+-- Developer
+INSERT INTO `permissions` (`code`, `description`, `categoryID`)
+VALUES ('issue_tracker_developer', 'Can re-assign issues, mark issues as closed, comment on all, edit issues.', 2);
+INSERT IGNORE INTO user_perm_rel (`userID`, `permID`) VALUES (
+  (SELECT ID FROM users WHERE UserID = 'admin'), (SELECT permID FROM permissions WHERE code = 'issue_tracker_developer')
+);
+
+-- LorisMenuPermissions
+INSERT INTO LorisMenuPermissions (MenuID, PermID)
+  SELECT m.ID, p.PermID FROM permissions p CROSS JOIN LorisMenu m WHERE p.code='issue_tracker_reporter' AND m.Label='Issue Tracker';
+
+-- Issues Table
+CREATE TABLE `issues` (
+  `issueID` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `title` varchar(255) NOT NULL DEFAULT '',
+  `reporter` varchar(255) NOT NULL DEFAULT '',
+  `assignee` varchar(255) DEFAULT NULL,
+  `status` enum('new','acknowledged','feedback','assigned','resolved','closed') NOT NULL DEFAULT 'new',
+  `priority` enum('low','normal','high','urgent','immediate') NOT NULL DEFAULT 'low',
+  `module` int(10) unsigned DEFAULT NULL,
+  `category` enum('Behavioural Instruments','Behavioural Battery','Data Entry','Database Problems','Examiners','SubprojectID/Project/Plan Changes') DEFAULT NULL,
+  `dateCreated` datetime DEFAULT NULL,
+  `lastUpdate` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  `lastUpdatedBy` varchar(255) DEFAULT NULL,
+  `sessionID` int(10) unsigned DEFAULT NULL,
+  `centerID` tinyint(2) unsigned DEFAULT NULL,
+  `candID` int(6) DEFAULT NULL,
+  PRIMARY KEY (`issueID`),
+  KEY `fk_issues_1` (`reporter`),
```

Comment:
Suggest using short format since it's way easier to read (maybe just my personal preference)
```
  PRIMARY KEY (`issueID`),
  FOREIGN KEY (`reporter`) REFERENCES `users` (`UserID`),
  FOREIGN KEY (`assignee`) REFERENCES `users` (`UserID`),
  FOREIGN KEY (`candID`) REFERENCES `candidate` (`CandID`),
  FOREIGN KEY (`sessionID`) REFERENCES `session` (`ID`),
  FOREIGN KEY (`CenterID`) REFERENCES `psc` (`CenterID`),
  FOREIGN KEY (`lastUpdatedBy`) REFERENCES `users` (`UserID`),
  FOREIGN KEY (`module`) REFERENCES `LorisMenu` (`ID`)
```